Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -260,7 +260,7 @@ public final class PackageBuilder {
 
     /// Computes the special directory where targets are present or should be placed in future.
     private func findTargetSpecialDirs(_ targets: [Target]) -> (targetDir: String, testTargetDir: String) {
-        let predefinedDirs = findPredefinedTargetDirectory()
+        let predefinedDirs = findPredefinedTargetDirectory(for: manifest.package.targets.filter { $0.path == nil })
 
```

Comment:
I feel that it's wrong to use here targets from the manifest instead of using targets that are passed into parameters.  I'm glad to hear your minds on this